pkg/kube/pod_command_executor.go

```diff
@@ -44,19 +62,29 @@ type podCommandExecutor struct {
 // Exec runs the command and logs stdout and stderr.
 func (p *podCommandExecutor) Exec(ctx context.Context, command []string, stdin io.Reader, stdout, stderr io.Writer) error {
 	var (
-		opts = ExecOptions{
+		stderrTail = NewLogTail(logTailDefaultLength)
+		stdoutTail = NewLogTail(logTailDefaultLength)
+		opts       = ExecOptions{
 			Command:       command,
 			Namespace:     p.namespace,
 			PodName:       p.podName,
 			ContainerName: p.containerName,
 			Stdin:         stdin,
-			Stdout:        stdout,
-			Stderr:        stderr,
+			Stdout:        stdoutTail,
+			Stderr:        stderrTail,
```

Comment:
LogTail collects only last lines of log, and keeps them internally. If error occurs, we create error which contains log tail. Thus, whether or not the caller passed the stdout/stderr writers, in case of error he will get an access to last log lines.

Reply:
thank you. What about 
> I just wanted to know, do we not use this way to execing a command into a pod when we use kube.Exec API?

Reply:
Sorry, forgot to answer this part - nope, currently we are not using and it's only part of PodController / PodCommandExecutor way of execution. Because for proper utilization of such error handling, the invoker should be prepared for such kind of error. But of course we can extend kube.Exec invokers and handle such erros.

Reply:
Right, I think we should do the same kind of handling even for `kube.exec` to make sure we are not seeing diff kinds of behaviour between creating a new pod and executing command in that and executing command in an already created pod.
Should we create an issue to track this?

Reply:
Probably you are right. Should I create an issue or you will do it ?

Reply:
https://kasten.atlassian.net/browse/K10-16438

Reply:
https://github.com/kanisterio/kanister/issues/2066